frontend/src/components/dashboard/create/email/EmailTemplate.tsx

```diff
@@ -30,29 +35,59 @@ const EmailTemplate = ({
   const bodyPlaceholder =
     'Dear {{ name }}, your next appointment at {{ clinic }} is on {{ date }} at {{ time }}'
 
-  async function handleSaveTemplate(): Promise<void> {
-    setErrorMsg(null)
-    try {
-      if (!campaignId) {
-        throw new Error('Invalid campaign id')
+  const handleSaveTemplate = useCallback(
+    async (next = true): Promise<void> => {
+      setErrorMsg(null)
+      try {
+        if (!campaignId) {
+          throw new Error('Invalid campaign id')
+        }
+        const { updatedTemplate, numRecipients } = await saveTemplate(
+          +campaignId,
+          subject,
+          body,
+          replyTo
+        )
+        onNext(
+          {
+            subject: updatedTemplate?.subject,
+            body: updatedTemplate?.body,
+            replyTo: updatedTemplate?.reply_to,
+            params: updatedTemplate?.params,
+            numRecipients,
+          },
+          next
+        )
+      } catch (err) {
+        setErrorMsg(err.message)
+
+        // Rethrow error to be handled elsewhere
+        if (!next) {
+          throw err
```

Comment:
The error will be handled in the modal and displayed as a message. Example of when this will happen is when the user try to save a Message A draft without `{{protectedlink}}`. 

<img src='https://user-images.githubusercontent.com/3666479/90603741-4e677f00-e22e-11ea-875f-2bab97bb01dd.png' width=500 />

Reply:
Ah I see thanks for clarifying. Nitpicking here; when i click on 'Save draft' and an error pops up I am very tempted to just skip saving draft, but that's probably just me.

Reply:
Do you think it would help if the "Save draft" button becomes "Edit draft" when there's an error? Currently to get back to the draft, the user will have to click on the close button at the top right of the modal.

Reply:
Yep that sounds great, I was thinking maybe an extra button "Edit draft" below the error message could work too ( and probably cleaner/ less work)